### ISSUE TITLE
Force Layout Zoom Dragging Problem => caused by simulation.find

### ISSUE THREAD

**Issue by patricknee** [d3/d3-force#83]
I am building a zoomable force layout on canvas with V4 in Angular. Sample project is available here:

https://github.com/patricknee/d3-angular2-force-drag-zoom-pan

When zoomed into the network, a dragged node does not follow the mouse; rather it travels at (mouse speed x zoom factor). 

Debugging this with single node examples, it appears that this occurs when the drag.subject() is located by simulation.find() but does not occur when using a home-grown find algorithm (which is adapted from the Drag + Zoom block and is similar to simulation.find() implementation).

In the flow drag().subject().dragStart(), if simulation.find is used, the d3.event.x (and y) is already transformed, while in the Drag+Zoom example and after implementing its homemade find(), the d3.event.x represents the canvas location and is ready to have transform.invertX() applied. 

In short, it appears that using simulation.find() changes the d3.event.x, preventing the correct dragging in the zoomed network. 

On the other hand, the homespun find() workaround introduces drawing problems I haven't yet resolved, but the node tracks the mouse correctly.

My repo includes multiple views and a complete demo of the problem. 

The project is based on two examples: [Drag + Zoom] (https://bl.ocks.org/mbostock/2b534b091d80a8de39219dd076b316cd) and [Force Dragging III] (http://bl.ocks.org/mbostock/ad70335eeef6d167bc36fd3c04378048), which has no zoom. These two base projects were reimplemented in Angular 2 to ensure the basic implementation is valid.

Thanks for any and all help!

**Comment by mbostock** [comment]
The drag behavior has no way of knowing that the canvas is zoomed, so I’m not sure there’s anything the library can do automatically for you here; you’ll need to transform the dragged coordinates manually, I expect.

It can handle this automatically in the case of SVG because as the transform attribute of the G element is modified, it affects the behavior of *element*.getScreenCTM (used by [d3.mouse](https://github.com/d3/d3-selection/blob/master/README.md#mouse) *etc.*)

**Comment by patricknee** [comment]
Thanks. Any idea why d3.event.[point] in the drag.subject() event handler represents the canvas coordinates, but in drag.start() event handler it is transformed, but this only occurs when simulation.find() is called rather than a home-grown version?

**Comment by mbostock** [comment]
When the *drag*.subject accessor is invoked, d3.event represents the “initiating input event” which should be either a native [*mousedown*](https://developer.mozilla.org/en-US/docs/Web/Events/mousedown) or [*touchstart*](https://developer.mozilla.org/en-US/docs/Web/Events/touchstart) event.

When the *start* event is dispatched, d3.event is a drag [*start* event](https://github.com/d3/d3-drag/blob/master/README.md#drag-events).

These two events are different, and expose different fields with different meanings.

**Comment by patricknee** [comment]
OK, I think I have it resolved.... doing some testing. I'll update my repo if it works. 

Thanks! Greatly appreciated.

**Comment by patricknee** [comment]
Problem resolved. I updated my repo so the working result can be seen by others.

Thanks again!
